### PR TITLE
feat(ios): support custom image as privacy screen

### LINKED ISSUE #27: feat: use splash screen
**Is your feature request related to a problem? Please describe.**
Currently a gray screen is displayed. It would be nice to show the splash screen of the app instead.

**Describe the solution you'd like**
Show the splash screen of the app instead of a gray screen.
The implementation of the Capacitor SplashScreen plugin should be used if possible.

**Describe alternatives you've considered**
No alternative solutions.

### PR DESCRIPTION
**Pull request by UlricW**
### Description
This pull request introduces a feature enhancement to the `capacitor-privacy-screen` plugin for iOS. The current behavior on iOS displays a grey background screen when the privacy screen is enabled. This pull request enhances the user experience by allowing the use of a custom background image instead of the default grey background. The background of the image (if the image doesn't take all the device screen) will be the system background (dependant of the dark theme is activated or not)

### Changes Made
- Added a new configuration options in the plugin settings specifically for iOS: 
    - `useImageBackground`. This option enables the user to specify a custom image file to be displayed as the background 
    -  `imageName`. This is the name of the image in iOS assets
- Modified the plugin's iOS implementation to load and display the custom background image when the privacy screen is enabled and the useImageBackground is true.

### Usage Example
```javascript

// Configure the plugin with custom background image
plugins: {
    PrivacyScreen: {
      enable: true,
      useImageBackground: true,
      imageName: "Splashscreen"
    },
  }

Close #27

### CODE REVIEW

**Comment by robingenz** [comment]
@UlricW Please merge the current main branch into your PR. I have created a separate implementation class because the plugin is getting more complex.

**Comment by UlricW** [comment]
[robingenz](https://github.com/robingenz) it's done

File: ios/Plugin/PrivacyScreenConfig.swift [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import Foundation
 
 public struct PrivacyScreenConfig {
-    var enable = true
+    var enable = true;
+    var useImageBackground = false;
```
Comment (robingenz) [inline ios/Plugin/PrivacyScreenConfig.swift:5]: This option can be removed or? If `imageName` is set, then it should be used.

Comment (robingenz) [inline ios/Plugin/PrivacyScreenConfig.swift:5]: Please add the type definitions for this option to the `definitions.ts` file and run `npm run docgen`.

**Review by robingenz — CHANGES_REQUESTED** [review]

File: ios/Plugin/PrivacyScreenConfig.swift (hunk 1 above)
Comment (UlricW) [reply ios/Plugin/PrivacyScreenConfig.swift]: Yes it can be removed, I updated the code to just check the `imageName`

Comment (UlricW) [reply ios/Plugin/PrivacyScreenConfig.swift]: It's done :)

**Comment by robingenz** [comment]
@UlricW Please make sure that the pipeline checks are successful.

**Comment by UlricW** [comment]
It's fixed, sorry for that :)

**Comment by robingenz** [comment]
@UlricW Thank you, looks great! I just had to release a hotfix. Please merge this again into your branch and then we are done. 🙌

**Comment by UlricW** [comment]
The merge is done !  :)

**Review by robingenz — APPROVED** [review]

**Comment by robingenz** [comment]
@UlricW Thank you for your contribution!

**Comment by subhashjha35** [comment]
Hi @robingenz 
Recently I stumbled upon this library as Privacy Screen was not functioning for android through AppStateChange. I wonder if we can have the same custom image support for android too to avoid having grey screen for content protection?

**Comment by robingenz** [comment]
@subhashjha35 On Android, this plugin sets the [FLAG_SECURE](https://developer.android.com/reference/android/view/WindowManager.LayoutParams#FLAG_SECURE) flag to treat the content of the window as secure, preventing it from appearing in screenshots or from being viewed on non-secure displays. As far as I know, there is no way to set a custom image. But feel free to create a PR if you find an option.

**Pull request merged**